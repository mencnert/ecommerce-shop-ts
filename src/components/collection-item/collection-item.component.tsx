import React from "react";

import "./collection-item.styles.scss";

export type Item = {
  id: number;
  name: string;
  price: number;
  imageUrl: string;
};

export const CollectionItem: React.FC<Item> = ({ name, price, imageUrl }) => (
  <div className="collection-item">
    <div className="image" style={{ backgroundImage: `url(${imageUrl})` }} />
    <div className="collection-footer">
      <span className="name">{name}</span>
      <span className="price">{price}</span>
    </div>
  </div>
);
