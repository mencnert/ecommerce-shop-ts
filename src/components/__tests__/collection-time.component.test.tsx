import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { CollectionItem } from "../collection-item/collection-item.component";

it("should match snapshot", () => {
  const data = {
    id: 1,
    name: "hello",
    price: 20,
    imageUrl: "url",
  };
  const wrapper = shallow(<CollectionItem {...data} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
