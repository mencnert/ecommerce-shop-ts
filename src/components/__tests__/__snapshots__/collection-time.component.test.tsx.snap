// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="collection-item"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(url)",
      }
    }
  />
  <div
    className="collection-footer"
  >
    <span
      className="name"
    >
      hello
    </span>
    <span
      className="price"
    >
      20
    </span>
  </div>
</div>
`;
